### PR TITLE
qemu: switch to emulated pflash to support UEFI vars

### PR DESCRIPTION
**Pull request by jakogut**
QEMU is able to optionally use OVMF firmware, built from Tianocore edk2,
to support UEFI. QEMU can be told to load OVMF firmware using a number
of methods, not all of which fully support persistent UEFI variables,
such as are required by secure boot.

Switch from using the -bios flag to fully emulated pflash to load OVMF,
including support for persistent UEFI variables. Also switch from the pc
machine to q35, which is required for fully emulated pflash.

https://github.com/tianocore/edk2/blob/e1e7306b54147e65cb7347b060e94f336d4a82d2/OvmfPkg/README#L65

Change-type: patch
Signed-off-by: Joseph Kogut <joseph@balena.io>

### CODE REVIEW

**Comment by klutchell** [comment]
Could you ELI5 what we gain from this PR?

**Comment by jakogut** [comment]
@klutchell This is required to support secure boot, which requires full pflash emulation for persistent variables. The keys used to verify the secure boot binaries (bootloader, kernel, etc.) are registered in the OVMF firmware menu, which saves them to the OVMF_VARS file. With this change, we can optionally specify a secure boot enabled OVMF firmware binary along with a variable file with our key registered, and test secure boot in the QEMU worker.

**Comment by klutchell** [comment]
Sounds good to me, feel free to merge if it's non-breaking!

**Comment by jakogut** [comment]
@balena-ci I self-certify!

**Comment by klutchell** [comment]
@balena-ci rebase
